HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -155,6 +155,11 @@ def customiseFor37046(process):
             delattr(prod, 'PrescaleCSVFile')
     return process
 
+def customizeForMuonPropagation(process):
+	process.load("TrackPropagation.SteppingHelixPropagator.SteppingHelixPropagatorAny_cfi")
+	process.load("TrackPropagation.SteppingHelixPropagator.SteppingHelixPropagatorAlong_cfi")
+	process.load("TrackPropagation.SteppingHelixPropagator.SteppingHelixPropagatorOpposite_cfi") 
+	return process
```

Comment:
Why is loading all these needed? Are these for various possible options but only one is used in the end?

Reply:
The HLT menu already contains:
```
process.SteppingHelixPropagatorAny = cms.ESProducer( "SteppingHelixPropagatorESProducer",
process.hltESPFastSteppingHelixPropagatorAny = cms.ESProducer( "SteppingHelixPropagatorESProducer",
process.hltESPFastSteppingHelixPropagatorOpposite = cms.ESProducer( "SteppingHelixPropagatorESProducer",
process.hltESPSteppingHelixPropagatorAlong = cms.ESProducer( "SteppingHelixPropagatorESProducer",
process.hltESPSteppingHelixPropagatorOpposite = cms.ESProducer( "SteppingHelixPropagatorESProducer",
```

Reply:
Perhaps some cleanup and rationalisation is needed!

Reply:
I did make the label if the propagators configurable so the ones already in the menu can be used. 
There is indeed some logic that chooses the propagator that is used in the code: https://github.com/cms-sw/cmssw/blob/23e7243726886e847f16ee3fd1dd4a4bc2f44884/MuonAnalysis/MuonAssociators/src/PropagateToMuonCore.cc#L148-L162

From what I can see, only the `Along` and `Opposite` are actually being used, so the `Any` could in principle be left out.